modules/nf-core/cellranger/multi/templates/cellranger_multi.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+"""
+Automatically rename staged files for input into cellranger multi and run it.
+
+Copyright (c) Felipe Almeida 2024 - MIT License
+"""
+from subprocess import run
+from pathlib import Path
+from textwrap import dedent
+import shlex
+import re
+
+
+def chunk_iter(seq, size):
+    """iterate over `seq` in chunks of `size`"""
+    return (seq[pos : pos + size] for pos in range(0, len(seq), size))
+
+
+work_dir  = str(Path.cwd())
+sample_id = "${prefix}"
+
+# get fastqs, ordered by path. Files are staged into
+#   - "fastq_001/{original_name.fastq.gz}"
+#   - "fastq_002/{original_name.fastq.gz}"
+#   - ...
+# Since we require fastq files in the input channel to be ordered such that a R1/R2 pair
+# of files follows each other, ordering will get us a sequence of [R1, R2, R1, R2, ...]
+fastqs = sorted(Path(".").glob("fastqs/*/[!EMPTY]*"))
+assert len(fastqs) % 2 == 0
+
+# target directory in which the renamed fastqs will be placed
+fastq_all = Path("./fastq_all")
+fastq_all.mkdir(exist_ok=True)
+
+# Match R1 in the filename, but only if it is followed by a non-digit or non-character
+# match "file_R1.fastq.gz", "file.R1_000.fastq.gz", etc. but
+# do not match "SRR12345", "file_INFIXR12", etc
+filename_pattern =  r'([^a-zA-Z0-9])R1([^a-zA-Z0-9])'
+
+for i, (r1, r2) in enumerate(chunk_iter(fastqs, 2)):
```

Comment:
enumeration should start at 1 (my bad actually), otherwise we'll have `L000` instead of `L001` as first file.

Reply:
Hi @grst ,
I have modified this one and I think it might be resolved now.